src/reformed.js

```diff
@@ -45,11 +46,26 @@ const makeWrapper = (middleware) => (WrappedComponent) => {
       }
     }
 
-    bindInput = (name) => {
+    bindInput = (name, opts) => {
+      const cfg = assign({
+        updateOn: 'onChange',
+        debounce: 0
+      }, opts)
+
+      const { updateOn } = cfg
+      const asyncHandler = (fn) => (e) => {
+        e.persist()
+        fn(e)
+      }
+
+      const updater = (!cfg.debounce)
+        ? this.bindToChangeEvent
+        : asyncHandler(debounce(this.bindToChangeEvent, cfg.debounce, false))
+
       return {
         name,
-        value: this.state.model[name] || '',
-        onChange: this.bindToChangeEvent,
+        defaultValue: this.state.model[name] || '',
```

Comment:
I think this requires some more thinking since `value` will block changes when `{ updateOn: 'onBlur' }`, but `defaultValue` does not apply the localStorage values.